src/sentry/notifications/manager.py

```diff
@@ -234,6 +236,12 @@ def _filter(
         if target_ids:
             filters["target_id__in"] = target_ids
 
+        if user_ids:
+            filters["user_id__in"] = user_ids
+
+        if team_ids:
+            filters["team_id__in"] = team_ids
```

Comment:
+1
Does this require verifying that all settings have either `user_id` or `team_id`?

Also does this require verifying that all calls to this method provide either user_ids or team_ids and not both?
Otherwise this would need to construct an `or` query, i.e. `sentry_notificationsetting.user_id in {user_ids} OR sentry_notificationsetting.team_id in {team_ids}`

Reply:
>  Does this require verifying that all settings have either user_id or team_id?

Yes, these changes are blocked by the backfill for notification settings.

> Also does this require verifying that all calls to this method provide either user_ids or team_ids and not both?

Yes. I can add that in.